src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.FunctionalTest/tests/GoToImplementation.test.ts

```diff
@@ -0,0 +1,97 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as assert from 'assert';
+import { afterEach, before, beforeEach } from 'mocha';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import {
+    mvcWithComponentsRoot,
+    pollUntil,
+    waitForProjectReady,
+} from './TestUtil';
+
+let cshtmlDoc: vscode.TextDocument;
+let editor: vscode.TextEditor;
+let cshtmlPath: string;
+
+suite('Implementation', () => {
+    before(async () => {
+        await waitForProjectReady(mvcWithComponentsRoot);
+    });
+
+    beforeEach(async () => {
+        cshtmlPath = path.join(mvcWithComponentsRoot, 'Views', 'Home', 'Index.cshtml');
+        cshtmlDoc = await vscode.workspace.openTextDocument(cshtmlPath);
+        editor = await vscode.window.showTextDocument(cshtmlDoc);
+    });
+
+    afterEach(async () => {
+        await vscode.commands.executeCommand('workbench.action.revertAndCloseActiveEditor');
+        await pollUntil(() => vscode.window.visibleTextEditors.length === 0, 1000);
+    });
+
+    test('Implementation of javascript returns nothing', async () => {
```

Comment:
Eh lets not test features we specifically disable. Not worth the functional test time spent.

Reply:
Well the concern here is that something incorrect gets out in this case, leading to opening some weird projected file, or that an error is thrown. Since the user can still press `Ctrl-F12` on a VS variable it's not so much that this features is specifically disabled as that the correct behavior is to return nothing.

Regardless, I know for now that we're doing to correct behavior, so I can still remove this if you'd like.

Reply:
That's definitely something that's good to verify but I'm hesitant to verify that in a functional test. Granted we don't have a way to unit test this code without a decent amount of refactoring.